src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -98,6 +98,66 @@ import { QuestionnaireForm } from "./QuestionnaireForm";
 import { QuestionnaireProperties } from "./QuestionnaireProperties";
 import ValueSetSelect from "./ValueSetSelect";
 
+const MAX_FILE_SIZE = 10 * 1024 * 1024; // 10MB in bytes
+
+interface ValidationResult {
+  isValid: boolean;
+  error?: string;
+  data?: QuestionnaireDetail;
+}
+
+const validateFileSize = (size: number): boolean => {
+  return size <= MAX_FILE_SIZE;
+};
+
+const validateQuestionnaireImport = (content: string): ValidationResult => {
+  try {
+    const data = JSON.parse(content);
+
+    if (!data.title || !Array.isArray(data.questions)) {
```

Comment:
good start, but what about other fields/also fields inside data.questions.

You can leverage Zod and existing validation logic.